### ISSUE TITLE
Shell crash on ctrl-r ctrl-c enter

### ISSUE THREAD

**Issue by nico** [SerenityOS/serenity#3413]
Repro:

1. Ctr-r
2. Ctrl-c
3. Enter

```
USERSPACE(42) ASSERTION FAILED: m_ptr
.././AK/RefPtr.h:224
[Shell(42:42)]: Terminating Shell(42) due to signal 6
[FinalizerTask(3:3)]: 0xdeadc0de  Kernel::Processor::switch_context(Kernel::Thread*&, Kernel::Thread*&) +181
0xdeadc0de  Kernel::Scheduler::context_switch(Kernel::Thread*) +183
0xdeadc0de  Kernel::Scheduler::pick_next() +1681
0xdeadc0de  Kernel::Processor::check_invoke_scheduler() +125
0xdeadc0de  Kernel::Thread::die_if_needed() +157
0xdeadc0de  syscall_handler +1414
0xdeadc0de  syscall_asm_entry +49
0x08084209  raise +22
0x080480bd  abort +13
0x0807fed7  tolower +0
0x0807c5f7  Line::Editor::enter_search() +91
0x08073f90  AK::Function<bool (Line::Editor&)>::CallableWrapper<Line::Editor::set_default_keybinds()::{lambda(auto:1&)#14}>::call(Line::Editor&) const +14
0x08077eec  Line::Editor::handle_read_event() +1184
0x08078311  AK::Function<void ()>::CallableWrapper<Line::Editor::get_line(AK::String const&)::{lambda()#1}>::call() const +43
0x080728bc  Core::Object::dispatch_event(Core::Event&, Core::Object*) +74
0x0806bcf0  Core::EventLoop::pump(Core::EventLoop::WaitMode) +308
0x0806be6f  Core::EventLoop::exec() +43
0x080792d9  Line::Editor::get_line(AK::String const&) +997
0x080653a7  Shell::read_single_line() +77
0x08065519  Shell::custom_event(Core::CustomEvent&) +29
0x080728bc  Core::Object::dispatch_event(Core::Event&, Core::Object*) +74
0x0806bcf0  Core::EventLoop::pump(Core::EventLoop::WaitMode) +308
0x0806be6f  Core::EventLoop::exec() +43
0x08048aa5  main +1725
0x08048f9e  _start +94
```

ctrl-r seems fairly broken in general, it doesn't find stuff.

**Comment by nico** [comment]
@alimpfard

**Comment by alimpfard** [comment]
That "reset suggestions" thing broke a lot of functionality huh...
At least, I assume that's the offending change?

**Comment by alimpfard** [comment]
> ctrl-r seems fairly broken in general, it doesn't find stuff

I didn't see a case of this, but I didn't check too much.

What I tried:
- last command
- sequence of `ls`'s with different options
- a `clear` somewhere in the middle of those `ls`'s
- some random `echo`'s I had in my history running on Lagom

These seem to be working ok?

**Comment by nico** [comment]
I tried `#asdf<enter><ctrl-r>as` and it didn't find anything but maybe it doesn't find comments?

**Comment by alimpfard** [comment]
Looks like it only matches from the beginning of the entries.
That's a silly oversight...

**Comment by nico** [comment]
Ah, I think all my local tests searched for stuff not at the start, that explains it.

**Comment by alimpfard** [comment]
Well, that's also fixed by #3415, so yay
